src/slice/mod.rs

```diff
@@ -115,6 +131,37 @@ pub trait ParallelSlice<T: Sync> {
             rem: snd,
         }
     }
+
+    /// Returns a parallel iterator over `N`-element chunks of
+    /// `self` at a time. The chunks do not overlap.
+    ///
+    /// If `N` does not divide the length of the slice, then the
+    /// last up to `N-1` elements will be omitted and can be
+    /// retrieved from the remainder function of the iterator.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rayon::prelude::*;
+    /// let chunks: Vec<_> = [1, 2, 3, 4, 5].par_array_chunks().collect();
+    /// assert_eq!(chunks, vec![&[1, 2], &[3, 4]]);
+    /// ```
+    fn par_array_chunks<const N: usize>(&self) -> ArrayChunks<'_, T, N> {
+        assert_ne!(N, 0);
+        let slice = self.as_parallel_slice();
+        let len = slice.len() / N;
+        let (fst, snd) = slice.split_at(len * N);
+        // SAFETY: We cast a slice of `len * N` elements into
+        // a slice of `len` many `N` elements chunks.
+        let array_slice = unsafe {
+            let ptr = fst.as_ptr() as *const [T; N];
+            ::std::slice::from_raw_parts(ptr, len)
+        };
```

Comment:
Oof, this is .. clever. I mean it makes sense, but for some reason it was hurting my head, I think because I was thinking of this as a `&[&[T]]` but of course that's not what we have here, it's a `&[[T;N]]`.

Reply:
FWIW, we did it the same way in the standard library [`array_chunks`](https://github.com/rust-lang/rust/pull/74373/files#diff-e8ccaf64ce21f955ccebef33b52158631493a6f0966815a2ebc142d7cd2b5e06R873-R881) and [`array_chunks_mut`](https://github.com/rust-lang/rust/pull/75021/files#diff-e8ccaf64ce21f955ccebef33b52158631493a6f0966815a2ebc142d7cd2b5e06R1067-R1077).